### PR TITLE
Restore RazorCompiler EA layer for desktop

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

File: src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler</RootNamespace>
+    <TargetFrameworks>$(SourceBuildTargetFrameworks);netstandard2.0</TargetFrameworks>
```
Comment (jjonescz) [inline src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj:7]:
```suggestion
    <TargetFrameworks>$(NetRoslynSourceBuild);netstandard2.0</TargetFrameworks>
```

File: src/Tools/ExternalAccess/RazorCompilerTest/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.UnitTests</RootNamespace>
+    <TargetFrameworks>net8.0;net472</TargetFrameworks>
```
Comment (jjonescz) [inline src/Tools/ExternalAccess/RazorCompilerTest/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.UnitTests.csproj:7]:
```suggestion
    <TargetFrameworks>$(NetRoslyn);net472</TargetFrameworks>
```

File: src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj
Comment (jjonescz) [inline src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj:1]: The title says "Restore RazorCompiler EA layer *for desktop*" - do we somehow limit this to be deployed only for net framework?

**Review by jjonescz — APPROVED** [review]

File: src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj
Comment (jaredpar) [reply src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj:1]: That's correct. There is no need for this in the .NET SDK right now.

Comment (jjonescz) [reply src/Tools/ExternalAccess/RazorCompiler/Microsoft.CodeAnalysis.ExternalAccess.RazorCompiler.csproj:1]: Ah, so we simply don't include this in transport packages to SDK, I assume.

**Review by RikkiGibson — APPROVED** [review]
🧟

**Pull request merged**